src/memory/AmdFlash.hh

```diff
@@ -85,6 +88,16 @@ public:
 		}
 	};
 
+	struct Erase final {
```

Comment:
Is `Erase` meant to be used in some class hierarchy? If not then the `final` annotation is confusing (to me). It's not wrong, but it also doesn't bring any benefits.

Same remark for `EraseOperation` below.

Reply:
For me `final` indicates exactly that it’s not meant to be used in a class hierarchy, preventing such use…

Reply:
OK fine. Keep it if you think it's important to explicitly indicate that.

OTOH, by far most structs/classes are not meant to be used in a class hierarchy. And c++ is already verbose enough.

Reply:
I’ll remove it. I thought it would help give the compiler cues for optimisation, but the absence of virtual pretty much already does that.